central/image/service/service_impl.go

```diff
@@ -476,9 +477,15 @@ func (s *serviceImpl) GetImageVulnerabilitiesInternal(ctx context.Context, reque
 	return internalScanRespFromImage(img), nil
 }
 
+func (s *serviceImpl) releaseFromScanSemaphore(n int64) {
+	metrics.ImageScanSemaphoreQueueSize.Add(float64(-n))
+	s.internalScanSemaphore.Release(n)
+}
+
 func (s *serviceImpl) acquireScanSemaphore(ctx context.Context) error {
 	semaphoreCtx, cancel := context.WithTimeout(ctx, maxSemaphoreWaitTime)
 	defer cancel()
+	metrics.ImageScanSemaphoreQueueSize.Add(1)
 	if err := s.internalScanSemaphore.Acquire(semaphoreCtx, 1); err != nil {
 		wrappedErr := errors.Wrap(err, "acquiring scan semaphore")
```

Comment:
Consider incrementing the count after `Acquire` is successful.

In the scenario where the Acquire was unsuccessful, the count will be increased but not decreased (`s.releaseFromScanSemaphore(1)` would not be called)

Reply:
I've refactored the whole metric usage, I think the first one I tried made little sense.

Reply:
Ahh - I see now,  the intent was to capture # of requests waiting.